File: solutions/ruby/anagram/1/anagram.rb
=begin
Write your code for the 'Anagram' exercise in this file. Make the tests in
`anagram_test.rb` pass.

To get started with TDD, see the `README.md` file in your
`ruby/anagram` directory.
=end

class Anagram
  def initialize(word)
    @word = word.downcase
  end

  def match(candidates)
    wordList = @word.chars.sort
    anagrams = Array.new
    for candidate in candidates
      if candidate.length != @word.length || candidate.downcase == @word.downcase
        next
      end
      candidateList = candidate.downcase.chars.sort
      if wordList.eql?(candidateList)
        anagrams.push(candidate)
      end
    end
    anagrams
  end
end

File: solutions/python/anagram/1/anagram.py
def find_anagrams(word, candidates):
    wordList = list(word.lower())
    wordList.sort()
    anagrams = []
    for candidate in candidates:
        if len(candidate) != len(word) or candidate.lower() == word.lower():
            continue
        candidateList = list(candidate.lower())
        candidateList.sort()
        matches = True
        for i in range(len(word)):
            if not wordList[i] == candidateList[i]:
                matches = False
                continue
        if matches:
            anagrams.append(candidate)
    return anagrams